lib/elm-test.js

```diff
@@ -661,13 +661,17 @@ function generateAndRunTests(tests, filePathArgs, generatedSrc, getGlobs) {
       };
     }
 
-    var watchedSourcePaths = fs
-      .readJsonSync(path.join(originalDir, "elm-package.json"), "utf8")
-      ["source-directories"].map(resolveWatchPath(originalDir));
+    var watchedSourcePaths;
+    var elmJson = fs.readJsonSync(path.join(projectRootDir, "elm.json"), "utf8");
+    if (elmJson["type"] === "package") {
```

Comment:
It might be worth moving all this kind of logic to a function. So instead we have `var watchedSourcePaths = getWatchedSourcePaths(elmJson['type'])`;

Reply:
Cleaned up in 7a530fc